monai/apps/auto3dseg/data_analyzer.py

```diff
@@ -234,6 +251,7 @@ def get_all_case_stats(self):
         if not self._check_data_uniformity([ImageStatsKeys.SPACING], result):
             logger.warning("Data is not completely uniform. MONAI transforms may provide unexpected result")
 
-        ConfigParser.export_config_file(result, self.output_path, fmt="yaml", default_flow_style=None)
+        if self.export:
+            ConfigParser.export_config_file(result, self.output_path, fmt="yaml", default_flow_style=None)
```

Comment:
this is hard-coded to yaml and a style, perhaps we make it a separate method and support more options cc @mingxin-zheng

Reply:
Agree. I should've done that earlier.

Reply:
@mingxin-zheng, let me address this with this PR though to avoid merge conflicts later on.